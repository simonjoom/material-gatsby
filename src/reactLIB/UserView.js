import React, { Component } from 'react';
import PropTypes from 'prop-types';
import LogoSkiscool from "../components/logo";

export const UserShape = {
  background: PropTypes.string,
  image: PropTypes.string,
  name: PropTypes.string,
  email: PropTypes.string
};

export class UserView extends Component {
  render() {
    const { background, image, name, email } = this.props;
    return (
      <div className="user-view">
        <LogoSkiscool/>
        {image && (
          <a href="#!user">
            <img className="circle" src={image} />
          </a>
        )}
        {name && (
          <a href="#!name">
            <span className="white-text name">{name}</span>
          </a>
        )}
        {email && (
          <a href="#!email">
            <span className="white-text email">{email}</span>
          </a>
        )}
      </div>
    );
  }
}

UserView.propTypes = UserShape;

export default UserView;
