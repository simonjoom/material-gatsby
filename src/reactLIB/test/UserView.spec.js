import React from 'react';
import { shallow } from 'enzyme';
import UserView from '../src/UserView';

describe('<UserView />', () => {
  let wrapper;

  test('renders', () => {
    wrapper = shallow(
      <UserView>
        <p>hello</p>
      </UserView>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
