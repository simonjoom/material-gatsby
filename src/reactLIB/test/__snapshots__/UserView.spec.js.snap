// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserView /> renders 1`] = `
<div
  className="user-view"
/>
`;
